aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,194 @@
+  * Accelerate app development.
+  * Reduce build pipeline complexity.
+  * Simplify maintenance.
+  * Leverage the existing .NET ecosystem of [.NET libraries](/dotnet/standard/class-libraries).
+  * Let developers understand and work on both client-side and server-side code.
+* Create reusable, sharable UI components.
+* Quickly get productive with Blazor reusable UI components from [top component vendors](https://blazor.net).
+* Work with all modern web browsers, including mobile browsers. Blazor uses open web standards without plug-ins or code transpilation.
+
+For more information, see <xref:blazor/components/prerendering-and-integration>.
+
+#### Blazor hosting options:
+
+Razor components have the flexibility of being hosted either using Blazor Server or Blazor WebAssembly.
+
+Blazor WebAssembly is a [single-page app (SPA) framework](/dotnet/architecture/modern-web-apps-azure/choose-between-traditional-web-and-single-page-apps) for building interactive client-side web apps that run in the browser on the [WebAssembly](https://webassembly.org/) .NET runtime (*Blazor WebAssembly*).
+
+Benefits of Blazor WebAssembly:
+
+* Uses open web standards without plugins or recompiling code into other languages.
+* Works with all modern web browsers, including mobile browsers.
+* WebAssembly code can access the full functionality of the browser via JavaScript, called *JavaScript interoperability*.
+
+To get started with ASP.NET Core Blazor WebAssembly, see <xref:blazor/tooling>. For an overview of ASP.NET Core Blazor WebAssembly, its architecture and benefits, see <xref:blazor/index#blazor-webassembly>.
+
+#### ASP.NET Core Blazor Server
+
+ASP.NET Core Blazor Server has the advantage of direct access to server resources, while providing model for building rich, interactive, and composable user interfaces in C#. Razor components in Blazor Server apps are UI elements such as a page, dialog, or data entry form, that handle client-side web UI interactions. The client-side web UI interactions are handled over a websocket connection and can span multiple connections to increase speed. Razor components keep track of the state of the client-side DOM and efficiently apply updates to only the components that need it.
+
+Blazor Server benefits:
+
+* Share app logic across server and client.
+* Flexibility of direct access to server resources. For example:
+  * Database.
+  * Secrets, such as access key-values used for API calls to [Azure Storage](/azure/storage/).
+* Uses a token-based security model, rather than using cookies. For example, cookies may not be enabled on the browser.
+
+To get started with ASP.NET Core Blazor Server, see [Get started with Blazor](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/intro). For an overview of ASP.NET Core Blazor Server, its architecture and benefits, see <xref:blazor/index>.
+
+### ASP.NET Core MVC plus hosted Blazor WebAssembly
```

Comment:
Earlier in the article it says that you can build server rendered UI, client rendered UI, or a hybrid of both. I think this section is addressing the hybrid use case. Perhaps it should go last then after SPA, which is a client scenario?

Also, this section only talks about Blazor WebAssembly, but you can render Blazor components from pages and views using Blazor Server too. Prerendering your components from a Blazor Server app is actually simpler because the components are already being run from the server. With Blazor WebAssembly prerendering is a bit trickier because you have to think about what it means to run the same component on both the server and client. But both scenarios are fully supported.

This section should link to https://docs.microsoft.com/aspnet/core/blazor/components/prerendering-and-integration